include/unifex/let_value_with_stop_source.hpp

```diff
@@ -131,48 +135,54 @@ class _stop_source_sender<SuccessorFactory>::type {
-        source_{source} {}
-    void operator()() noexcept {
-        source_.request_stop();
-    }
-    inplace_stop_source& source_;
-};
-
-template<typename SuccessorFactory, typename Receiver>
+template <typename SuccessorFactory, typename Receiver>
 struct _stop_source_operation<SuccessorFactory, Receiver>::type {
+  using stop_token_type = stop_token_type_t<Receiver>;
+  using stop_source_type = fused_stop_source<stop_token_type>;
+  using receiver_t = stop_source_receiver<
+      operation<SuccessorFactory, Receiver>,
+      remove_cvref_t<Receiver>>;
+  using inner_sender_t =
+      unifex::invoke_result_t<SuccessorFactory&&, stop_source_type&>;
 
-    template <typename SuccessorFactory2, typename Receiver2>
-    type(SuccessorFactory2&& func, Receiver2&& r) :
-        func_{(SuccessorFactory2&&)func},
-        stop_source_{},
-        // Chain the stop token so that downstream cancellation also affects this
-        // operation
-        stop_callback_(
-            unifex::get_stop_token(r),
-            _stop_source_operation_callback(stop_source_)),
-        innerOp_(
-              unifex::connect(
-                ((SuccessorFactory&&)func_)(stop_source_),
-                stop_source_receiver<operation<SuccessorFactory, Receiver>, remove_cvref_t<Receiver2>>{
-                    *this,
-                    static_cast<Receiver2&&>(r)})) {
-    }
-
-    void start() noexcept {
-        unifex::start(innerOp_);
-    }
+private:
+  static constexpr bool successor_is_nothrow =
+      unifex::is_nothrow_invocable_v<SuccessorFactory&&, stop_source_type>;
+  static constexpr bool inner_receiver_nothrow_constructible =
+      unifex::is_nothrow_constructible_v<receiver_t, type*, Receiver&&>;
+  static constexpr bool nothrow_connectable =
+      unifex::is_nothrow_connectable_v<inner_sender_t, receiver_t>;
+
+  auto connect_inner_op(SuccessorFactory& func, Receiver&& r) noexcept(
+      successor_is_nothrow && inner_receiver_nothrow_constructible &&
+          nothrow_connectable) {
+    return unifex::connect(
+        static_cast< SuccessorFactory&&>(func)(stopSource_),
+        receiver_t{*this, static_cast<Receiver&&>(r)});
+  }
 
-    template<class F>
-    using callback_type = typename stop_token_type_t<Receiver>::template callback_type<F>;
-    UNIFEX_NO_UNIQUE_ADDRESS SuccessorFactory func_;
-    UNIFEX_NO_UNIQUE_ADDRESS unifex::inplace_stop_source stop_source_;
-    UNIFEX_NO_UNIQUE_ADDRESS callback_type<_stop_source_operation_callback> stop_callback_;
-    connect_result_t<
-        callable_result_t<SuccessorFactory, unifex::inplace_stop_source&>,
-        stop_source_receiver<operation<SuccessorFactory, Receiver>, remove_cvref_t<Receiver>>>
-        innerOp_;
+public:
+  template <typename SuccessorFactory2, typename Receiver2>
+  type(SuccessorFactory2&& func, Receiver2&& r) noexcept(
+      is_nothrow_constructible_v<SuccessorFactory, SuccessorFactory2&&>&&
```

Comment:
Nit: the `&&` is redundant here.
```suggestion
      is_nothrow_constructible_v<SuccessorFactory, SuccessorFactory2>&&
```